backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/VdsCpuUnitPinningHelper.java

```diff
@@ -50,25 +45,19 @@ public boolean isDedicatedCpuPinningPossibleAtHost(Map<Guid, List<VdsCpuUnit>> v
 
         previewPinOfPendingExclusiveCpus(cpuTopology, vmToPendingPinnings, vm.getCpuPinningPolicy());
 
-        int vcpus = vm.getNumOfCpus();
-        int cpusLeft = vcpus;
-        int hostCoresPerSocket = host.getCpuCores() / host.getCpuSockets();
-        int hostThreadsPerCore = host.getCpuThreads() / host.getCpuCores();
+        int socketsLeft = vm.getNumOfSockets();
 
         for (int socket : getOnlineSockets(cpuTopology)) {
-            int coresInSockets = getCoresInSocket(cpuTopology, socket).size();
-            if (hostCoresPerSocket * hostThreadsPerCore == coresInSockets &&
-                    coresInSockets >= vcpus &&
-                    getFreeCpusInSocket(cpuTopology, socket).size() < vcpus) {
-                // we should fit into one socket
-                continue;
+            int coresInSocket = getAvailableCores(cpuTopology, socket, vm.getThreadsPerCpu());
+            while (socketsLeft > 0 && coresInSocket >= vm.getCpuPerSocket()) {
+                socketsLeft--;
+                coresInSocket -= vm.getCpuPerSocket();
```

Comment:
how about replacing that while with a computation that is similar to `socketsLeft -= coreCount / vm.getCpuPerSocket();` below? (it needs to take into account that coresInSocket may not be a multiply of vm.getCpuPerSocket() and may exceed it